library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsSampleStreamWrapper.java

```diff
@@ -853,8 +944,10 @@ private void initMediaChunkLoad(HlsMediaChunk chunk) {
     mediaChunks.add(chunk);
 
     chunk.init(this);
-    for (HlsSampleQueue sampleQueue : sampleQueues) {
+    for (int i = 0; i < sampleQueues.length; i++) {
+      HlsSampleQueue sampleQueue = sampleQueues[i];
       sampleQueue.setSourceChunk(chunk);
+      chunk.setFirstSampleIndex(i, sampleQueue.getWriteIndex());
```

Comment:
There is actually an edge case where chunk.shouldSpliceIn is true in which we can't use the write index as it is. This happens if we switch playlists for example. I changed the code to use the firstSampleIndices from the previous chunk instead to be on the safe side.